src/scripts/main.js

```diff
@@ -1,5 +1,60 @@
 'use strict';
 
 document.addEventListener('click', (e) => {
-  // write code here
+  // Getting wall and spider elements
+  const spider = document.querySelector('.spider');
+  const wall = document.querySelector('.wall');
+
+  // Getting spider size
+  const spiderHeight = spider.clientHeight;
+  const spiderWidth = spider.clientWidth;
+
+  // Getting wall size
+  const wallRect = wall.getBoundingClientRect();
+  const borderWidth = parseFloat(
+    window.getComputedStyle(wall).getPropertyValue('border-width'),
+  );
+  // Getting coordinates of the click relative to the wall
+  // const x = e.clientX - wallRect.left;
+  // const y = e.clientY - wallRect.top;
+
+  const x = e.pageX - wallRect.left;
+  const y = e.pageY - wallRect.top;
+
+  // Checking if the click is inside the wall boundaries considering the spider
+  if (x >= 0 && y >= 0 && x <= wallRect.width && y <= wallRect.height) {
+    // Calculating new coordinates for spider
+    let spiderX = x - spiderWidth / 2 - borderWidth;
+    let spiderY = y - spiderHeight / 2 - borderWidth;
+
+    // get possible spider placement rect
+    const availableAreaLeft = 0;
+    const availableAreaRight = wallRect.width - 2 * borderWidth - spiderWidth;
```

Comment:
The calculation for `availableAreaRight` subtracts `2 * borderWidth`, which is correct for the right side. Ensure that the border width is consistently accounted for in all calculations.